portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,4 +83,16 @@ private String convertToJsonUsingGson(List<String> messages) {
     String json = gson.toJson(messages);
     return json;
   }
+
+  private int getNumComments(HttpServletRequest request) {
+      String userInput = request.getParameter("max");
```

Comment:
Move "max" to a static const. We might change the parameter name in future and this should not break the server logic.